lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -6,13 +6,13 @@ class Service < KubernetesResource
     def sync(cache)
       super
       if exists? && selector.present?
-        @related_deployments = cache.get_all(Deployment.kind, selector)
-        @related_statefulsets = cache.get_all(StatefulSet.kind, selector)
         @related_pods = cache.get_all(Pod.kind, selector)
+        # Optimization: assume that the pod selector matches the workload selector
+        @related_workloads = workloads_fetcher(cache, selector)
```

Comment:
This 'optimization' wont always do the right thing. I suspect it needs to be removed